File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/CS/Shipper1/Program.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ServiceModel;
using System.ServiceModel.Description;
using System.Workflow.Runtime;

namespace Microsoft.WorkflowServices.Samples
{
    class Program
    {
        static void Main(string[] args)
        {
            WorkflowServiceHost host = new WorkflowServiceHost(typeof(ShipperWorkflow));
            host.Description.Behaviors.Find<WorkflowRuntimeBehavior>().WorkflowRuntime.WorkflowTerminated += delegate(object sender, WorkflowTerminatedEventArgs e) { Console.WriteLine("WorkflowTerminated: " + e.Exception.Message); };
            host.Description.Behaviors.Find<WorkflowRuntimeBehavior>().WorkflowRuntime.WorkflowCompleted += delegate(object sender, WorkflowCompletedEventArgs e) { Console.WriteLine("WorkflowCompleted."); };
            host.Open();

            Console.ForegroundColor = ConsoleColor.White;
            Console.WriteLine("Role: Shipper");
            Console.WriteLine("Press <enter> to exit");
            Console.ResetColor();
            Console.ReadLine();
            host.Close();
        }
    }
}

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/VB/Shipper1/Program.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

#Region "Using directives"

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Threading
Imports System.Workflow.Runtime
Imports System.Workflow.Runtime.Hosting
Imports System.ServiceModel
Imports System.ServiceModel.Description


#End Region

Namespace Microsoft.WorkflowServices.Samples
	Class Program
        Shared Sub Main(ByVal args As String())
            Dim host As New WorkflowServiceHost(GetType(ShipperWorkflow))
            AddHandler host.Description.Behaviors.Find(Of WorkflowRuntimeBehavior)().WorkflowRuntime.WorkflowTerminated, AddressOf WorkflowRuntime_WorkflowTerminated
            AddHandler host.Description.Behaviors.Find(Of WorkflowRuntimeBehavior)().WorkflowRuntime.WorkflowCompleted, AddressOf WorkflowRuntime_WorkflowCompleted
            host.Open()

            Console.ForegroundColor = ConsoleColor.White
            Console.WriteLine("Press <enter> to exit")
            Console.ResetColor()
            Console.ReadLine()
            host.Close()
        End Sub
        Private Shared Sub WorkflowRuntime_WorkflowTerminated(ByVal sender As Object, ByVal e As WorkflowTerminatedEventArgs)
            Console.WriteLine("WorkflowTerminated: " + e.Exception.Message)
        End Sub
        Private Shared Sub WorkflowRuntime_WorkflowCompleted(ByVal sender As Object, ByVal e As WorkflowCompletedEventArgs)
            Console.WriteLine("WorkflowCompleted.")
        End Sub
	End Class
End Namespace